packages/turf-helpers/index.ts

```diff
@@ -598,15 +598,20 @@ export function multiPolygon<P extends GeoJsonProperties = GeoJsonProperties>(
  * // => collection
  */
 export function geometryCollection<
+  T extends
+    | Point
+    | LineString
+    | Polygon
+    | MultiPoint
+    | MultiLineString
+    | MultiPolygon,
   P extends GeoJsonProperties = GeoJsonProperties,
 >(
-  geometries: Array<
-    Point | LineString | Polygon | MultiPoint | MultiLineString | MultiPolygon
-  >,
+  geometries: Array<T>,
   properties?: P,
   options: { bbox?: BBox; id?: Id } = {}
-): Feature<GeometryCollection, P> {
-  const geom: GeometryCollection = {
+): Feature<GeometryCollection<T>, P> {
+  const geom: GeometryCollection<T> = {
```

Comment:
This change has an interesting backstory.

packages/turf-line-chunk/types.ts has `lineChunk(geomCollection, 2);`, but `geometryCollection` was erasing the specific subtype (the default is just `GeometryCollection<Geometry>`) instead of inferring them forward into the return type. This changes `geometryCollection` to preserve any sort of information we did have for the things getting put into the collection.

I believe that this class of type strictness error may hit other consumers if they are creating geometry collections using other methods.